### PR TITLE
Imp 012 fix render

### PR DESCRIPTION
**Pull request by jackyzha0**
cHONK

### CODE REVIEW

File: IMP/board.py [hunk 1]
```diff
@@ -13,37 +15,53 @@ def __init__(self, w, h, foodspawn = 0.1):
         self.generation = 0  # current generation
         self.food = []  # initialized later
         self.creatures = []  # initialized later
-        self.GEN_TIMEOUT = 1800  # Constant for generation timeout
+        self.GEN_TIMEOUT = 12000  # Constant for generation timeout
         self.FOOD_GEN_DELTA = 5  # Constant for food generation position change
+        self.MAX_FOOD = 500
+
+        pygame.init()
+        self.BACKGROUND_COLOR = (220,220,220) # white
```
Comment (scarletnguyen13) [inline IMP/board.py]: fix the color comment - this is not white anymore

**Review by chenanton — COMMENTED** [review]
looks good to me.

File: IMP/board.py [hunk 2]
```diff
@@ -13,37 +15,53 @@ def __init__(self, w, h, foodspawn = 0.1):
...
-            for i2 in c.collide(self.creatures[i+1:]):
-                c2 = self.creatures[i2]       # get creature from list using index
-
-                if c.size > c2.size:          # check which creature is bigger
-                    self.creatures.pop(i2)    # kill the smaller one
-                    c.energy += c2.size       # increase the bigger one's energy by the smaller one's size
+            for col_creat in c.collide(self.creatures[i+1:]):
+                if c.size > col_creat.size:          # check which creature is bigger
+                    self.creatures.remove(col_creat)    # kill the smaller one
+                    c.energy += col_creat.size * 5       # increase the bigger one's energy by the smaller one's size
```
Comment (scarletnguyen13) [inline IMP/board.py]: We should have a variable for the 5

File: IMP/board.py [hunk 3]
```diff
@@ -13,37 +15,53 @@ def __init__(self, w, h, foodspawn = 0.1):
...
                 else:
                     self.creatures.pop(i)
-                    c2.energy += c.size
+                    col_creat.energy += c.size
+            for f in c.collide(self.food):
+                if c.size > f.size:          # check if can eat
+                    self.food.remove(f)
+                    c.energy += f.size
+                    self.g_l[i].fitness += f.size * 5
```
Comment (scarletnguyen13) [inline IMP/board.py]: We should have a variable for the 5

File: IMP/board.py [hunk 4]
```diff
@@ -13,37 +15,53 @@ def __init__(self, w, h, foodspawn = 0.1):
...
+
+        if self.ticks_total % 2 == 0:
```
Comment (scarletnguyen13) [inline IMP/board.py]: We also should have a variable for the 2

**Review by scarletnguyen13 — APPROVED** [review]
LLLLGGGTTTMMM 👯‍♂

**Pull request merged**